extension/src/runtime.ts

```diff
@@ -0,0 +1,30 @@
+import * as vscode from "vscode";
+import * as fs from "fs";
+import * as path from "path";
+
+const xcrExtension = ".xcr";
+
+export const launchRuntime = (terminals: { [key: string]: vscode.Terminal }): void => {
+    const rootPath = vscode.workspace.rootPath;
+    const config = vscode.workspace.getConfiguration();
+    const xcruntimePath = (fs.existsSync(config.xcruntime.path)) ? config.xcruntime.path : "xcruntime.exe";
+    const isWindowsPlatform = /^win/.test(process.platform);
+    const mono = (isWindowsPlatform) ? "" : "mono";
+    const xcassemblies = `${rootPath}${path.sep}XCR${path.sep}xcassemblies`;
+    let xcrFileFound = false;
+    fs.readdirSync(xcassemblies).forEach(file => {
```

Comment:
this code is doing two things at the same time... if looks for a file and it  creates a terminal.
break it in two, first look for a file, then if it is found, ccreate the terminal.

just to avoid unintended behaviour if somehow we find more than one xcr in the folder.